ocs_ci/utility/users.py

```diff
@@ -144,13 +179,9 @@ def _finalizer():
         Delete all users created by the factory
 
         """
-        with open(htpasswd_path) as f:
-            htpasswd = f.readlines()
-        new_htpasswd = [line for line in htpasswd if not line.startswith(tuple(_users))]
-        with open(htpasswd_path, "w+") as f:
-            for line in new_htpasswd:
-                f.write(line)
-        create_htpasswd_secret(htpasswd_path, replace=True)
```

Comment:
I think that the secret needs to be updated too. According to documentation the step of updating the secret is in place and following commands to delete user and identity takes place only when the secret is updated.

Reply:
https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html-single/authentication_and_authorization/index#identity-provider-htpasswd-update-users_configuring-htpasswd-identity-provider

Reply:
sorry... I did not get you...

Reply:
You are removing a step that updates the secret. It needs to stay to have users fully deleted - it is one of the steps from the documentation that is linked (step 3). You should not remove these lines.